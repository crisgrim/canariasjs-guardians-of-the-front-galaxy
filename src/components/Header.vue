<template>
  <div class="nav">
    <router-link to="/">
      Home
    </router-link>
    <router-link to="/about">
      About
    </router-link>
  </div>
</template>

<script>
export default {
  name: 'Header',
};
</script>

<style scoped lang="scss">
.nav {
  background-color: #3ab981;
  display: flex;
  align-items: flex-start;
  justify-content: flex-start;
  padding: 10px;
}

.nav a {
  color: #ffffff;
  padding-right: 10px;
  text-decoration: none;
}
</style>
