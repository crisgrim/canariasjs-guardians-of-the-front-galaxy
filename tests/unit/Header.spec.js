import { shallowMount, RouterLinkStub } from '@vue/test-utils';
import Header from '../../src/components/Header.vue';

describe('Header.spec.js', () => {
  const cmp = shallowMount(Header, {
    stubs: {
      RouterLink: RouterLinkStub,
    },
  });

  it('has the expected html structure', () => {
    expect(cmp.element).toMatchSnapshot();
  });

  it('has path properly', () => {
    expect(cmp.find(RouterLinkStub).props().to).toBe('/');
  });
});
