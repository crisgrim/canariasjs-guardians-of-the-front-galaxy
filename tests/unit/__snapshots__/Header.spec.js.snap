// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.spec.js has the expected html structure 1`] = `
<div
  class="nav"
>
  <a>
    
    Home
  
  </a>
   
  <a>
    
    About
  
  </a>
</div>
`;
